### PR TITLE
EDU-868 Keep pre-determined media duration on reset

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-865

### CODE REVIEW

File: src/components/media-player/media-player-track-group.js [hunk 1]
```diff
@@ -29,25 +29,36 @@ const createInitialTrackStates = sources => ({
...
+
+const hasAnyMediaChanged = (currentTrackStates, newSources) => {
+  return hasMainTrackMediaChanged(currentTrackStates, newSources)
     || newSources.secondaryTracks.length !== currentTrackStates.secondaryTracks.length
-  ) {
-    return true;
+    || currentTrackStates.secondaryTracks.some((track, index) => hasSecondaryTrackMediaChanged(currentTrackStates, newSources, index));
+};
+
+const resetTrackStates = (currentTrackStates, sources) => {
```
Comment (adrianaluchian) [inline src/components/media-player/media-player-track-group.js:48]: A bit of help and context on these changes:

When **any** track changes its media data (`sourceUrl` or `playbackRange`), we were resetting the  `trackState`. This meant the `duration` was lost for **all** tracks too. However the `duration` only gets determined again if the `sourceUrl` of a track is changed.

And due to this code
```
    if (!!mainTrackDuration && secondaryTrackDurations.every(x => !!x)) {
      onDuration(mainTrackDuration);
    }
```
we only called the `onDuration` callback when all durations were there.

So basically for the situation in the JIRA ticket, this was never called after the range was changed.

So within the reset of the `trackStates`, I keep the `durations` if the track "media" data is not changed. Everything else makes sense to be reset and kept in sync between the tracks.

Comment (ahelmberger) [reply src/components/media-player/media-player-track-group.js:48]: Thanks, this info was very helpful!

**Review by ahelmberger — APPROVED** [review]
